spectractor/extractor/background.py

```diff
@@ -214,8 +220,13 @@ def extract_spectrogram_background_sextractor(data, err, ws=(20, 30), mask_signa
                        filter_size=(filter_size, filter_size),
                        sigma_clip=sigma_clip, bkg_estimator=bkg_estimator,
                        mask=mask)
-    bgd_model_func = interp2d(np.arange(Nx), np.arange(Ny), bkg.background, kind='linear', bounds_error=False,
-                              fill_value=None)
+    bgd_model_func_interp = RegularGridInterpolator((np.arange(Nx), np.arange(Ny)), bkg.background.T, method='linear',
+                                                    bounds_error=False, fill_value=None)
+
+    def bgd_model_func(x, y):
+        xx, yy = np.meshgrid(x, y, indexing='ij')
+        return bgd_model_func_interp((xx, yy)).T
```

Comment:
Ok, I put these lines inside a external function, but not sure that it solves the problem of "local function" because I need to still need to return a function in a function...

Reply:
Well, it's a step in the right direction at least, so let's say that's OK for now 🙂